### PR TITLE
[GEOT-6533] Selection of source matrixset when cascading WMTS layers does not take into account Longitudefirst CRSs

### PR DESCRIPTION
**Pull request by singhdan**
- Changed the way in which the source matrixset is selected so that it compares the request SRS and the matrixset SRS names rather than comparing the whole CRS. This will reduce the amount of transformation that will be needed when the tiles are returned
- Added unit tests to verify that the changes work as expected
- Some spelling corrections

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 

- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [ ] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [X] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [X] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [X] PR for bug fixes and small new features are presented as a single commit
- [X] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [X] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: modules/extension/wmts/src/test/java/org/geotools/ows/wmts/WebMapTileServerTest.java [hunk 1]
```diff
@@ -16,21 +16,27 @@
...
+import org.geotools.ows.wmts.request.GetTileRequest;
 import org.geotools.referencing.CRS;
+import org.geotools.tile.Tile;
 import org.junit.Test;
 import org.opengis.referencing.crs.CoordinateReferenceSystem;
 
 /** @author Emanuele Tajariol (etj at geo-solutions dot it) */
 public class WebMapTileServerTest {
 
+    private static PrintStream out = System.out;
```
Comment (aaime) [inline modules/extension/wmts/src/test/java/org/geotools/ows/wmts/WebMapTileServerTest.java]: Tests should not be writing to system.out, use the logging framework instead, or if you need it for development, keep the line that does the printout commented out, using System.out directly.

File: modules/extension/wmts/src/main/java/org/geotools/ows/wmts/request/AbstractGetTileRequest.java [hunk 2]
```diff
@@ -351,12 +351,15 @@ private TileMatrixSet selectMatrixSet() throws ServiceException, RuntimeExceptio
...
-        // If the layer provides the requested CRS, use it
+        // See if the layer supports the requested SRS. Matching against the SRS rather than the
+        // full CRS will avoid missing matches due to Longitude first being set on the request
+        // CRS and therefore minimise transformations that need to be performed on the received tile
         for (TileMatrixSet matrixSet : capabilities.getMatrixSets()) {
 
             CoordinateReferenceSystem matrixCRS = matrixSet.getCoordinateReferenceSystem();
-
-            if (CRS.equalsIgnoreMetadata(requestCRS, matrixCRS)) { // matching SRS
+            String requestSRS = CRS.toSRS(requestCRS);
```
Comment (aaime) [inline modules/extension/wmts/src/main/java/org/geotools/ows/wmts/request/AbstractGetTileRequest.java]: No need to convert it at every round in the loop, could be done outside.

**Comment by taba90** [comment]
> Had a look but I'm not familiar with this module. I've asked @taba90 to double check, he's been playing with the module lately.

Tested this interactively and wmts works as expected.

File: modules/extension/wmts/src/test/java/org/geotools/ows/wmts/WebMapTileServerTest.java (hunk 1 above)
Comment (singhdan) [reply modules/extension/wmts/src/test/java/org/geotools/ows/wmts/WebMapTileServerTest.java]: Removed all print statements. I only needed them for development and no need for them to be left in and commented out

File: modules/extension/wmts/src/main/java/org/geotools/ows/wmts/request/AbstractGetTileRequest.java (hunk 2 above)
Comment (singhdan) [reply modules/extension/wmts/src/main/java/org/geotools/ows/wmts/request/AbstractGetTileRequest.java]: Agreed, moved to outside the loop

**Pull request merged**